packages/components/src/box-control/test/index.js

```diff
@@ -192,4 +192,117 @@ describe( 'BoxControl', () => {
 			} );
 		} );
 	} );
+
+	describe( 'Unit selections', () => {
+		it( 'should update unlinked controls unit selection based on all input control', () => {
+			// Render control
+			render( <BoxControl /> );
+
+			// Make unit selection on all input control
+			const allUnitSelect = screen.getByLabelText( 'Select unit' );
+			allUnitSelect.focus();
+			fireEvent.change( allUnitSelect, { target: { value: 'em' } } );
+
+			// Unlink the controls
+			const unlink = screen.getByLabelText( /Unlink Sides/ );
+			fireEvent.click( unlink );
+
+			// Confirm that each individual control has the selected unit
+			const unlinkedSelects = screen.getAllByDisplayValue( 'em' );
+			expect( unlinkedSelects.length ).toEqual( 4 );
+		} );
+
+		it( 'should use individual side attribute unit when available', () => {
+			// Render control
+			const { rerender } = render( <BoxControl /> );
+
+			// Make unit selection on all input control
+			const allUnitSelect = screen.getByLabelText( 'Select unit' );
+			allUnitSelect.focus();
+			fireEvent.change( allUnitSelect, { target: { value: 'vw' } } );
+
+			// Unlink the controls
+			const unlink = screen.getByLabelText( /Unlink Sides/ );
+			fireEvent.click( unlink );
+
+			// Confirm that each individual control has the selected unit
+			const unlinkedSelects = screen.getAllByDisplayValue( 'vw' );
+			expect( unlinkedSelects.length ).toEqual( 4 );
+
+			// Rerender with individual side value & confirm unit is selected.
+			rerender( <BoxControl values={ { top: '2.5em' } } /> );
+
+			const topSelect = screen.getByDisplayValue( 'em' );
+			const otherSelects = screen.getAllByDisplayValue( 'vw' );
+
+			expect( topSelect ).toBeInTheDocument();
+			expect( otherSelects.length ).toEqual( 3 );
+		} );
+	} );
+
+	describe( 'onChange updates', () => {
+		it( 'should call onChange when values contain more than just CSS units', () => {
+			const setState = jest.fn();
+
+			render( <BoxControl onChange={ setState } /> );
+
+			const input = screen.getByLabelText( 'Box Control', {
+				selector: 'input',
+			} );
+
+			input.focus();
+			fireEvent.change( input, { target: { value: '7.5rem' } } );
+			fireEvent.keyDown( input, { keyCode: ENTER } );
+
+			expect( setState ).toHaveBeenCalledWith( {
+				top: '7.5rem',
+				right: '7.5rem',
+				bottom: '7.5rem',
+				left: '7.5rem',
+			} );
+		} );
+
+		it( 'should not pass invalid CSS unit only values to onChange', () => {
+			const setState = jest.fn();
+
+			render( <BoxControl onChange={ setState } /> );
+
+			const allUnitSelect = screen.getByLabelText( 'Select unit' );
+			allUnitSelect.focus();
+			fireEvent.change( allUnitSelect, { target: { value: 'rem' } } );
+
+			expect( setState ).toHaveBeenCalledWith( {
+				top: undefined,
+				right: undefined,
+				bottom: undefined,
+				left: undefined,
+			} );
+		} );
+
+		it( 'should clear side attribute when control is cleared', () => {
```

Comment:
In this test case, the value is cleared by committing it via the ENTER key. Should we add another one for clearing the field, and then switching focus (tabbing or clicking out of it?)

Reply:
Thanks. I only added the test for committing the change via the `ENTER` key because I was leaving the behaviour for the field to revert to the original value on blur. 

If its agreed we should change that as well, I'll definitely add a test covering it. 👍

Reply:
Gotcha, that sounds good to me.

Reply:
This test is currently failing. Once https://github.com/WordPress/gutenberg/pull/33485 lands, we can then update the BoxControl to set `required={ false }` on its UnitControl components, which should then get this passing again, if I'm following correctly?

Reply:
I've removed that test in https://github.com/WordPress/gutenberg/pull/33444/commits/ca2fe1cfbc232da221bd079755c75d9d4aa6b74b.

If the current proposal in https://github.com/WordPress/gutenberg/pull/33485 is accepted we can add the test back along with the `required={ false }` prop, if needed. If the `NumberControl` ends up being changed so that it isn't required by default I'm not sure we need the test.